gbfs-validator/cli.js

```diff
@@ -1,17 +1,65 @@
 const { inspect } = require('util')
+const commander = require('commander')
+const fs = require('fs')
+const fsPath = require('path')
 const GBFS = require('./')
+const pjson = require('./package.json');
 
-if (!process.argv[2]) {
-  console.error('Usage: gbfs-validator GBFS_URL')
-  process.exit(1)
+getFeedValidationReport = async (url) => {
+  const gbfs = new GBFS(url)
+  return gbfs.validation()
 }
 
-const gbfs = new GBFS(process.argv[2])
+parseOptions = () => {
+  commander
+    .version(pjson.version, '-v, --version')
+    .usage('[OPTIONS]...')
+    .option('-u, --url <dataset_folder>', 'URL of the GBFS feed')
+    .option('-p, --print', 'Print report to console')
+    .option('-vb, --verbose', 'Verbose mode prints debugging console logs')
+    .option('-s, --save-report <report_path>', 'Local path to output report file')
+    .parse(process.argv)
 
-if (require.main === module) {
-  gbfs
-    .validation()
-    .then(result => console.log(inspect(result, { depth: null, colors: true })))
-} else {
-  module.exports = gbfs.validation()
+  return commander.opts()
+}
+
+const options = parseOptions();
+
+const saveReport = (report, filePath) => { 
+  const dirname = fsPath.dirname(filePath);
+  if (!fs.existsSync(dirname)) {
+    fs.mkdirSync(dirname, { recursive: true });
+  }
+  fs.writeFileSync(filePath, JSON.stringify(report))
 }
+
+const processFeedValidation = async () => {
+  if (options.verbose) {
+    console.log("Started GBFS validation with options: \n " + inspect(options, { depth: null, colors: true }))
+  }
+  try {
+    const report = await getFeedValidationReport(options.url)
+    if (options.print) {
+      console.log(inspect(report, { depth: null, colors: true }))
+    }
+    if (options.saveReport) {
+      saveReport(report, options.saveReport)
+    }
+  } catch (error) {
+    console.error(`Critical error while validating GBFS feed => ${error}`)
+    process.exit(1);
+  }
+}
+
+if (options.url) {
```

Comment:
As this parameter is mandatory, it may be preferable to pass the URL such as `node gbfs-validator/cli.js [URL]` instead of `-u`.

Reply:
Good point, I'll add the option as mandatory. I considered adding `-f` for local files in the _near_ future; that's why defining `-u` for URLs and `-f` for files gives consistency for the consumer.